activemodel/test/cases/errors_test.rb

```diff
@@ -411,6 +411,30 @@ def test_no_key
     assert_equal({ name: [{ error: :blank }, { error: :invalid }] }, person.errors.details)
   end
 
+  test "slice! removes the given keys from the errors" do
+    person = Person.new
+    person.errors.add(:name, "cannot be nil")
+    person.errors.add(:age, "cannot be nil")
+    person.errors.add(:gender, "cannot be nil")
+    person.errors.add(:city, "cannot be nil")
+
+    person.errors.slice!(:age, :gender)
+
+    assert_equal [:age, :gender], person.errors.keys
+  end
+
+  test "slice! returns the deleted errors" do
+    person = Person.new
+    person.errors.add(:name, "cannot be nil")
+    person.errors.add(:age, "cannot be nil")
+    person.errors.add(:gender, "cannot be nil")
+    person.errors.add(:city, "cannot be nil")
+
+    removed_errors = person.errors.slice!(:age, :gender)
```

Comment:
We need a test with string keys, since all other methods support both strings and symbols.

Reply:
Sure! 

Should I mix strings and symbols in this test or is it better to create another test just for that?

Reply:
You can mix them if you want.